### PR TITLE
Bump minimum pandas version to 0.22.0, switch from pd.util.testing to pd.testing

### LINKED ISSUE #901: switch from pandas.util.testing to pandas.testing
`pandas.util.testing` is deprecated and we're supposed to switch to the functions in the public API at `pandas.testing` instead. 

I'm not sure what the pandas version compatibility is for this. If there's a problem, we could use `try/except` statements to handle earlier versions of pandas. Or we could just do nothing until the min pandas we support is high enough.

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] Closes #901
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~Tests added~
 - ~Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~
 - ~Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).~
 - ~New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Only the `assert_*` functions are exposed in the public module `pandas.testing`, so I had to import the `network` decorator from `pandas._testing` instead.  I'm not sure why it isn't public, but its [docstring example](https://github.com/pandas-dev/pandas/blob/master/pandas/_testing.py#L2446) imports from `_testing` so I guess that's the way to do it.

`git grep -n 'util\.test'` only shows one remaining instance, an output cell in the tracking tutorial notebook that should be taken care of by #999. 

FYI there is a new deprecation warning:
```python
pvlib/iotools/ecmwf_macc.py:303: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    times = netCDF4.num2date(nctime[time_slice], nctime.units)
```
Looks like it only gets raised in our tests from netCDF4 calls.  See https://github.com/Unidata/netcdf4-python/issues/1023

### CODE REVIEW

**Comment by kanderso-nrel** [comment]
The python 3.5 checks are failing, I suppose because the latest pandas version available for 3.5 doesn't have `pandas._testing` for the `network` decorator.  Some ways forward:

- just stop using the network decorator.  Not sure what value it provides now anyway, unless people like to run the tests while not connected to the internet?
- drop support for 3.5 and require `pandas >= 1.0.0`
- change where `network` is imported from based on the pandas version
- hold off on merging this

**Comment by wholmgren** [comment]
> just stop using the network decorator. Not sure what value it provides now anyway, unless people like to run the tests while not connected to the internet?

my preference

**Comment by wholmgren** [comment]
It appears that none of the test failures are due to pandas min dependency not being increased, and it appears that we've lost the test configuration that used https://github.com/pvlib/pvlib-python/blob/master/ci/requirements-py35-min.yml 

I think we need to get that working again before we can take action on this PR.

**Comment by kanderso-nrel** [comment]
> it appears that we've lost the test configuration that used https://github.com/pvlib/pvlib-python/blob/master/ci/requirements-py35-min.yml

Do I understand correctly that py35-min is used on Travis but not Azure?  It seems like Travis is still running those tests, despite not posting the results to the PR: https://travis-ci.org/github/pvlib/pvlib-python/builds/708201652.  This same thing happened with RdTools too: https://github.com/NREL/rdtools/pull/138#issuecomment-597353176 -- maybe the travis yaml is out of date or something.

**Comment by kanderso-nrel** [comment]
Possibly relevant, but I think only someone with admin access to pvlib can do this: https://travis-ci.community/t/github-status-not-posted-on-commits-on-repositories-using-legacy-service-integration/7798

**Comment by kanderso-nrel** [comment]
After merging master so that python35-min is included, we do get failures on pandas 0.18.1: https://dev.azure.com/solararbiter/pvlib%20python/_build/results?buildId=3817&view=logs&j=b0b9a916-ca36-5efa-d29b-53c06cc9f57e&t=4c652da1-4a81-5d6f-5221-c0e955e0fbca&l=31

Should we bump pandas to 0.22 in this PR?

**Comment by wholmgren** [comment]
That works for me.

File: docs/sphinx/source/whatsnew/v0.8.0.rst [hunk 1]
```diff
@@ -57,6 +57,7 @@ Documentation
 
 Requirements
 ~~~~~~~~~~~~
+* minimum pandas version increased to v0.22.0, released Dec 31, 2017. (:pull:`1003`)
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.8.0.rst:60]:
```suggestion
* Minimum pandas version increased to v0.22.0, released Dec 31, 2017. (:pull:`1003`)
```

Comment (wholmgren) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:60]: I like the inclusion of the release date!

Comment (kanderso-nrel) [reply docs/sphinx/source/whatsnew/v0.8.0.rst]: Just copying @mikofski :)  https://github.com/pvlib/pvlib-python/blame/master/docs/sphinx/source/whatsnew/v0.7.0.rst#L214

**Pull request merged**